libraries/ms-helper/src/can_interval.c

```diff
@@ -0,0 +1,102 @@
+#include "can_interval.h"
+
+#include <stdint.h>
+
+#include "can.h"
+#include "can_uart.h"
+#include "generic_can.h"
+#include "objpool.h"
+#include "soft_timer.h"
+#include "status.h"
+
+static ObjectPool s_can_interval_pool;
```

Comment:
One approach I'm looking into for dealing with static storage is writing the interface to accept a pointer like our other modules, but providing a pointer to static storage that lives within the module. That would let us easily switch to another source while still allowing the primary static storage to exist.

Reply:
That's a good idea! Let me know when we have a paradigm flushed out and I'll help with converting existing code to use it.